lib/openstudio-standards/utilities/assertion.rb

```diff
@@ -40,12 +40,24 @@ def prm_get_optional_handler(component, log_dir, data_key, *remaining_keys)
   target_data = component.send(data_key)
   prm_raise(target_data.is_initialized,
             log_dir,
-            "Failed to retrieve data: #{data_key} from #{component.name.get}"
+            "Failed to retrieve data: #{data_key} from #{prm_get_component_name(component)}"
             )
   target_data_get = target_data.get
   return remaining_keys.length == 0 ? target_data_get : prm_get_optional_handler(target_data_get, log_dir, remaining_keys[0], *remaining_keys[1...])
 end
 
+# This is a PRM handler to get a name from an OpenStudio object instance
+# If the object instance does not have a name, then it will return the object name.
+#
+# @param component [OpenStudio] an OpenStudio object
```

Comment:
I'm not sure if what's in the `[]` is the proper way of documenting an OpenStudio object.

Reply:
I am not sure too. I checked the API, it seems Component might be the correct one:

[Component](https://s3.amazonaws.com/openstudio-sdk-documentation/cpp/OpenStudio-3.6.1-doc/model/html/classopenstudio_1_1model_1_1_component.html) is a [Model](https://s3.amazonaws.com/openstudio-sdk-documentation/cpp/OpenStudio-3.6.1-doc/model/html/classopenstudio_1_1model_1_1_model.html) that contains exactly one [ComponentData](https://s3.amazonaws.com/openstudio-sdk-documentation/cpp/OpenStudio-3.6.1-doc/model/html/classopenstudio_1_1model_1_1_component_data.html) object.

All objects other than the [ComponentData](https://s3.amazonaws.com/openstudio-sdk-documentation/cpp/OpenStudio-3.6.1-doc/model/html/classopenstudio_1_1model_1_1_component_data.html) object and the version object are the [Component](https://s3.amazonaws.com/openstudio-sdk-documentation/cpp/OpenStudio-3.6.1-doc/model/html/classopenstudio_1_1model_1_1_component.html) contents. There is a primary [ModelObject](https://s3.amazonaws.com/openstudio-sdk-documentation/cpp/OpenStudio-3.6.1-doc/model/html/classopenstudio_1_1model_1_1_model_object.html) that provides the [Component](https://s3.amazonaws.com/openstudio-sdk-documentation/cpp/OpenStudio-3.6.1-doc/model/html/classopenstudio_1_1model_1_1_component.html)'s type. There are no public constructors, as Components are to be constructed through the [ModelObject](https://s3.amazonaws.com/openstudio-sdk-documentation/cpp/OpenStudio-3.6.1-doc/model/html/classopenstudio_1_1model_1_1_model_object.html) method [createComponent](https://s3.amazonaws.com/openstudio-sdk-documentation/cpp/OpenStudio-3.6.1-doc/model/html/classopenstudio_1_1model_1_1_model_object.html#a1da254fca2b648896a8c418985d86598). Similarly, Model-level object addition and removal, and SqlFile functionality, has been disabled.